core/src/main/kotlin/fr/sncf/osrd/api/project_signals/SignalProjectionEndpoint.kt

```diff
@@ -33,8 +33,10 @@ class SignalProjectionEndpoint(private val infraManager: InfraProvider) : Take {
             val blockPath = infra.blockInfra.convertBlockPath(request.blocks)
             val routePath = infra.rawInfra.convertRoutePath(request.routes)
 
-            val signalProjections = mutableMapOf<Long, List<SignalUpdate>>()
-            for ((id, trainSimulation) in request.trainSimulations) {
+            val signalProjections =
```

Comment:
It seems like the order does not depend on anything but the order of `request.trainSimulations`. So you can simplify the code to:
```suggestion
            val signalProjections = mutableListOf<SignalUpdate>()
             for ((index, trainSimulation) in request.trainSimulations) {
                val signalProjection =
                    projectSignals(
                        infra,
                        chunkPath,
                        blockPath,
                        routePath,
                        trainSimulation.signalCriticalPositions,
                        trainSimulation.zoneUpdates,
                        trainSimulation.simulationEndTime
                    )
                signalProjections.add(signalProjection)
            }
```

Reply:
Done :)